src/intel_driver.c

```diff
@@ -56,6 +56,51 @@ uint32_t g_intel_debug_option_flags = 0;
 #define LOCAL_I915_PARAM_EU_TOTAL 34
 #endif
 
+bool intel_batchbuffer_configure_watchdog(int fd, unsigned int gem_ctx_id, int flag)
+{
+    unsigned threshold[5];
+    int ring_flag;
+
+    ring_flag = flag & I915_EXEC_RING_MASK;
+
+    struct drm_i915_gem_context_param cp = {
+        .ctx_id = gem_ctx_id,
+        .param = LOCAL_I915_CONTEXT_PARAM_WATCHDOG,
+        .value = (uint64_t)&threshold,
+        .size = sizeof(threshold)
+    };
+
+    memset(threshold, 0, sizeof(threshold));
+
+    /* set watchdog threshold to VCS's engine only */
+    if (ring_flag == I915_EXEC_BSD || ring_flag == LOCAL_I915_EXEC_VCS2) {
+        threshold[VIDEO_DECODE_CLASS] = 1200;
```

Comment:
Is 1200 suitable for all media workload?

Reply:
You're right, that's a guestimate for now, I'll try to come up with a better solution that could account for different workloads and different display resolutions, for example, a higher threshold at 4K...

Reply:
Would it be possible to do something based on frame rate? However, a) I don't know if that information is available at this level and b) it would probably require distinguishing between batch and real mode which I don't know if it is possible. But for real-time it certainly sounds plausible for a layman like me that if a frame hasn't been decoded in 1/fps time the stack might want to give up and do something else. What happens with inter-frame dependencies is another question I don't have the answer for.